Logic/PetBase.js

```diff
@@ -74,6 +77,15 @@ const initiatePet = (pet) => {
       pet.elementName = "dark"
       break
   }
+  for (let i = 0; i < PetProfessionBonusNames.length; i++) {
+    const valueArray = PetProfessionBonusNames[i].value.split(",")
+    for (const v of valueArray) {
+      if (v == pet.profBonus) {
+        pet.profBonusName = PetProfessionBonusNames[i].label
+        break
+      }
+    }
+  }
```

Comment:
You could just use the some method on the value array.

const petProfessionBonusValues = PetProfessionBonusNames[i].value.split(",");
if (petProfessionBonusValues.some(pet.profBonus)) {
    pet.profBonusName = PetProfessionBonusNames[i].label
}

Reply:
I think it's ok to be undefined in the results if we don't have an entry in the table for some reason.  Your "simpler" example sort of blows my mind as I'm a native C/C++ programmer by trade.  The first example is more readable to my less experienced eyes.  Your code base, so your preference.  I'll test both.

Reply:
From the JS documentation isn't `.some()` supposed to take a callback function?  Or is there a fallback for checking values when a non-function is passed in?

Reply:
I think it works with `indexOf()` instead of `some()`

Reply:
Something like this:
```
pet.profBonusName= PetProfessionBonusNames
  .find(bonus => bonus.value.split(",").indexOf(pb) > -1)?.label ?? ''
```

Reply:
(I had to read up on what the `?` were doing in that statement)

Reply:
Yeah I was a bit hasty. Probably smart to see my examples as pseudocode as I freehand it in comments without an IDE. it should be some(value => value === pet.profBonus) but indexOf works but I think it overall offers less clarity.
Some clearly stated it will be true if some match the condition.
The -1 in the indexOf function means it's not there so if it's more than -1 it's found. Bit more mental overhead for me personally but im very used to the functional type programming in js.

Either is fine, just like offering options.

Reply:
And the ? is a null-safe operator so if nothing is found instead of trying to access .label and throwing it will return undefined. It's called null safe but it checks for both null and undefined.
It should probably be after the find function as well as that could be undefined.

Reply:
Your example was good, this works like you said.
```
pet.profBonusName = PetProfessionBonusNames
  .find(bonus => bonus.value.split(",").indexOf(pet.profBonus) > -1)?.label ?? ''"
```

Reply:
I went with your call on the `some()` function:
```
  pet.profBonusName = PetProfessionBonusNames
    .find(bonus => bonus.value.split(",").some(value => value == pet.profBonus))?.label ?? 'Unknown Bonus'
```

NOTE: I used `==` for the compare due to datatype issues (I think the array has strings and the pet.profBonus ends up being a number).  If that's no good let me know.

Reply:
Sweet, looks good to me.

Reply:
I went with your call on the some() function:
```
  pet.profBonusName = PetProfessionBonusNames
    .find(bonus => bonus.value.split(",").some(value => value == pet.profBonus))?.label ?? 'Unknown Bonus'
```
NOTE: I used == for the compare due to datatype issues (I think the array has strings and the pet.profBonus ends up being a number). If that's no good let me know.